File: frontend/src/app/controller/service/admin/commun/ModeReglementAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {ModeReglementDto} from 'src/app/controller/model/commun/ModeReglement.model';
import {ModeReglementCriteria} from 'src/app/controller/criteria/commun/ModeReglementCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class ModeReglementAdminService extends AbstractService<ModeReglementDto, ModeReglementCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/modeReglement/');
    }

    public constrcutDto(): ModeReglementDto {
        return new ModeReglementDto();
    }

    public constrcutCriteria(): ModeReglementCriteria {
        return new ModeReglementCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/easystock/domain/service/facade/admin/commun/ModeReglementAdminService.java
package ma.zs.easystock.domain.service.facade.admin.commun;

import java.util.List;
import ma.zs.easystock.domain.model.commun.ModeReglement;
import ma.zs.easystock.infra.dao.criteria.core.commun.ModeReglementCriteria;
import ma.zs.easystock.zynerator.service.IService;



import ma.zs.easystock.ws.dto.commun.ModeReglementDto;
import org.springframework.http.HttpEntity;


public interface ModeReglementAdminService extends  IService<ModeReglement,ModeReglementCriteria>  {



    HttpEntity<byte[]> createPdf(ModeReglementDto dto) throws Exception;

}
